### PR TITLE
test: add unit testing

### PR DESCRIPTION
**Pull request by sgtcoolguy**
This adds some basic unit tests to the module, primarily verifying constants exist and their basic type.

Even these tests revealed documented constants that have since been removed (around SSO), and that iOS/Android differ in using numbers vs strings for some of the constants. As a result, we do have some tests failing here.

### CODE REVIEW

File: test/unit/specs/constants.spec.js [hunk 1]
```diff
@@ -6,13 +6,13 @@ const ANDROID = (Ti.Platform.osname === 'android');
...
 
 	it('can be required', () => {
-		Facebook = require('ti.facebook');
+		Facebook = require('facebook');
 		expect(Facebook).toBeDefined();
 	});
 
-	it('.apiName', () => {
-		expect(Facebook.apiName).toBe('Ti.Facebook');
-	});
```
Comment (hansemannn) [inline test/unit/specs/constants.spec.js]: `Facebook.moduleId` always equals the module ID, in this case `facebook` would pass properly.

File: test/unit/specs/module.spec.js [hunk 2]
```diff
@@ -0,0 +1,346 @@
...
+					});
+
+					it('PLACE_LOCATION_CONFIDENCE_HIGH', () => {
+						expect(Facebook.PLACE_LOCATION_CONFIDENCE_HIGH).toEqual(jasmine.any(Number));
+					});
+				});
+			}
+
+			describe('LOGIN_BEHAVIOR_*', () => {
+				// FIXME: On Android these are Strings, but in docs/iOS they're numbers!
```
Comment (janvennemann) [inline test/unit/specs/module.spec.js]: It looks like you already addressed this issue and changed everything to numbers, right?

File: package.json [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    "danger": "^7.1.0",
+    "eslint": "^5.10.0",
+    "eslint-config-axway": "^4.1.1",
+    "eslint-plugin-jasmine": "^2.10.1",
+    "fs-extra": "^7.0.1",
+    "jasmine-core": "^3.3.0",
+    "karma": "^4.0.0",
+    "karma-jasmine": "^2.0.1",
+    "karma-junit-reporter": "^1.2.0",
+    "karma-mocha": "^1.3.0",
```
Comment (janvennemann) [inline package.json]: There are some unused dependencies here:

- karma-mocha
- karma-webpack
- mocha
- webpack

**Review by janvennemann — DISMISSED** [review]
Only two minor notes

**Comment by janvennemann** [comment]
@sgtcoolguy What do we do with the failing tests on iOS? I think the issue can be fixed by using the `USE_VIEW_FOR_UI_METHOD` macro in the view proxy for now, for example:

```objc
USE_VIEW_FOR_UI_METHOD(tooltipBehavior)
```

*EDIT:* Seems our docs are outdated on that one (the macro does not exist anymore) and it didn't handle return values anyway. To actually fix it for now we could manually delegate the property access.

```objc
- (NSNumber *_Nonnull)tooltipBehavior
{
  return ((FacebookLoginButton *)self.view).tooltipBehavior;
}
```

File: test/unit/specs/module.spec.js (hunk 2 above)
Comment (sgtcoolguy) [reply test/unit/specs/module.spec.js]: yes, I did change these

**Pull request merged**